File: docs-examples/example-kotlin/src/main/kotlin/io/micronaut/configuration/openapi/docs/customserializers/MyJaxbElement.kt
package io.micronaut.configuration.openapi.docs.customserializers

// tag::clazz[]
// if you want to use generic from fields with type JAXBElement<T>
class MyJaxbElement<T> {
    var type: String? = null
    var value: T? = null
}
// end::clazz[]

File: docs-examples/example-python/src/main/python/micronaut/configuration/openapi/docs/customserializers/MyJaxbElement.py
from dataclasses import dataclass
from typing import Generic, TypeVar

from micronaut.core.annotation import Introspected

T = TypeVar("T")


# tag::clazz[]
# if you want to use generic from fields with type JAXBElement<T>
@Introspected
@dataclass
class MyJaxbElement(Generic[T]):
    type: str
    value: T
# end::clazz[]
